app/controllers/dragons_teams_controller.rb

```diff
@@ -36,12 +38,40 @@ def destroy
   def add_dragon
     @dragon = current_user.dragons.build(dragon_params)
     if @dragon.save
+      pay_for_dragon
       redirect_to dragons_teams_index_path(current_user.id)
```

Comment:
Zazwyczaj `redirect_to` i `flash[]` umieszcza się konsekwentnie albo w głównej metodzie kontrolera (index, update, create itp.) albo w prywatnej metodzie.

Tutaj jeśli smoków jest za dużo to jest w głównej metodzie:
```
if current_user.dragons.count >= 5
  flash[:alert] = "You can't add more dragons"
  redirect_to dragons_teams_index_path(current_user.id)
```

jeśli może kupić to w metodzie prywatnej, a jeśli nie to znowu w głównej metodzie.

Potem łatwo się w tym pogubić. Proponowałbym wszstkie `redirect_to` i `flash[]` przenieść do głównej metody